lnc/store.go

```diff
@@ -0,0 +1,31 @@
+package lnc
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+var (
+	// ErrSessionNotFound is returned when a session is not found in the
+	// database.
+	ErrSessionNotFound = errors.New("session not found")
+)
+
+// Store represents access to a persistent session store.
+type Store interface {
+	// AddSession adds a record for a new session in the database.
+	AddSession(ctx context.Context, session *Session) error
+
+	// GetSession retrieves the session record matching the passphrase
+	// entropy.
+	GetSession(ctx context.Context,
+		passphraseEntropy []byte) (*Session, error)
+
+	// SetRemotePubKey sets the remote public key for a session.
+	SetRemotePubKey(ctx context.Context, label string,
+		remotePubKey []byte) error
+
+	// SetExpiry sets the expiry time for a session.
+	SetExpiry(ctx context.Context, label string, expiry time.Time) error
```

Comment:
still getting through the diff but what happens when the session expires? does aperture shutdown?

Reply:
The next call will fail and the server will shutdown